kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -149,13 +150,13 @@ impl<'tcx> GotocHook<'tcx> for Assert {
 
         // Since `cond` might have side effects, assign it to a temporary
         // variable so that it's evaluated once, then assert and assume it
+        // TODO: I don't think `cond` can have side effects, this is MIR, it's going to be temps
```

Comment:
Can we just assert that? https://github.com/model-checking/kani/blob/632fda645364790ada91b5e9980731638c7fc8b7/cprover_bindings/src/goto_program/expr.rs#L316

Reply:
This code bothered me, but it's a distraction from the point of this PR, but I couldn't help but shove a comment in there for later...